internal/executor/fake/context/context.go

```diff
@@ -50,7 +50,8 @@ var DefaultNodeSpec = []*NodeSpec{
 type FakeClusterContext struct {
 	clusterId             string
 	pool                  string
-	handlers              []*cache.ResourceEventHandlerFuncs
+	podEventHandlers      []*cache.ResourceEventHandlerFuncs
+	clusterEventHandlers  []*cache.ResourceEventHandlerFuncs
```

Comment:
Maybe this doesn't matter since this is a fake executor, but `clusterEventHandlers` doesn't seem to be used anywhere, other than assignment. Is that expected?

Reply:
That is expected, I just added it so if anyone wanted to use it in future they don't make the same mistake

Reply:
Alternatively we could just do nothing on AddClusterEventEventHandler and remove that field

Reply:
I think it's fine to keep it.  Maybe just add a comment to say it's not used right now?

Reply:
It's fine, I just lack context around how the fake executor works. A comment would be nice but not necessary.